flexmeasures/api/dev/tests/test_sensor_data.py

```diff
@@ -0,0 +1,55 @@
+from flask import url_for
+import pytest
+
+from flexmeasures.api.tests.utils import get_auth_token
+from flexmeasures.api.dev.tests.utils import make_sensor_data_request
+
+
+@pytest.mark.parametrize("use_auth", [False, True])
+def test_post_sensor_data_bad_auth(client, use_auth):
+    """
+    Attempt to post sensor data with insufficient or missing auth.
+    """
+    # the case without auth: authentication will fail
+    headers = {"content-type": "application/json"}
+    if use_auth:
+        # in this case, we successfully authenticate,
+        # but fail authorization (no admin or MDC role)
+        headers["Authorization"] = get_auth_token(
+            client, "test_supplier@seita.nl", "testtest"
+        )
+
+    post_data_response = client.post(
+        url_for("post_sensor_data"),
+        headers=headers,
+    )
+    print("Server responded with:\n%s" % post_data_response.data)
+    if use_auth:
+        assert post_data_response.status_code == 403
+    else:
+        assert post_data_response.status_code == 401
+
+
+@pytest.mark.parametrize(
+    "request_field, new_value, error_field, error_text",
+    [
+        ("duration", "PT30M", "_schema", "Resolution of 0:05:00 is incompatible"),
```

Comment:
It is not evident to me why this resolution is incompatible. I guess it is because the sensor resolution is longer than 5 minutes, and we don't support downsampling? Please add a note to the test for future devs.